src/composeapp/appengine.cc

```diff
@@ -21,11 +21,11 @@ AppEngine::Result AppEngine::fetch(const App& app) {
     if (local_source_path_.empty()) {
       exec(boost::format{"%s --store %s pull -p %s --storage-usage-watermark %d"} % composectl_cmd_ % storeRoot() %
                app.uri % storage_watermark_,
-           "failed to pull compose app");
+           "failed to pull compose app", "", nullptr, "", true);
```

Comment:
Does the `timeout` tool break `print_output = true`?
Having some timeout set may serve as a safety net in case of something bad happens inside `composectl` which might lead to it hanging indefinitely. So I would keep it if possible.

Reply:
No, it does not break the `print_output = true`.

The `composectl pull` has its own timeouts, so this external timeout can interfere with the the command internal timeouts.
Also, the pull can take hours under poor bandwidth. We can set `timeout` to a few hours to make it safe for the pull command, not sure it is worth it though.

Reply:
I think a few hours is better than nothing

Reply:
Set 4 hour timeouts for both `pull` and `install` commands, see this commit d79a3f9a.